### PR TITLE
[ext] Connect the extension to the banners API

### PR DESCRIPTION
**Pull request by sigmike**
**Related issue** #1769

---

### Description

The extension now gets the banner content from the API.

On the home page, on dark mode, with a standard banner:

image

On the search page, on light mode, with a security advisory banner:

image

Other changes:

- I made the `fetchTournesolApi` more flexible and more similar to `fetch` by making it take only a path and an object of options. (43ed52398bfc84c63769bdbbe3ab2bb58a3edc7d)
- The banner's close button is always aligned to the right.
- A red warning icon is displayed when the banner is a security advisory (instead of the megaphone icon). The icon is taken from MUI's icons[1].
- The TournesolRecommendations and TournesolSearchRecommendations are now only instantiated when the user is on the right page. This was necessary to avoid the systematic double loading of the banners, as these instances require a Banner that is already loaded.
- When the user closes the banner the preference is stored with a key which derived from the name of the banner (`displayBanner:${name}`).
- The title of the banner is displayed above the text.
- The language of the banner is determined by the Accept-Language header, which is automatically set by the browser. Firefox sends its UI language setting, not the language setting of the page content.
- The action button won't be displayed if there is no action link or label.
- The alt text of the icons have been translated into French.

[1] https://github.com/mui/material-ui/blob/b62d5757e6dc218c57cf6db77dea78c7b2396a5a/packages/mui-icons-material/material-icons/warning_24px.svg

### Checklist

- [x] I added the related issue(s) id in the related issues section (if any)
- [x] I described my changes and my decisions in the PR description
- [x] I read the development guidelines of the [CONTRIBUTING.md][development-guidelines]
- [x] The tests pass and have been updated if relevant
- [x] The code quality check pass

### CODE REVIEW

**Review by GresilleSiffle — APPROVED** [review]
Hello @sigmike 

Thanks a lot for your work.

In addition to the minor suggestions, I was wondering: do you think it would be a good idea to align the banner's action button to the right, next to the close button?

Currently, the button is aligned to the left with the text. It's not a blocking point, but it's a bit surprising.

**Comment by sigmike** [comment]
> In addition to the minor suggestions, I was wondering: do you think it would be a good idea to align the banner's action button to the right, next to the close button?
> 
> Currently, the button is aligned to the left with the text. It's not a blocking point, but it's a bit surprising.

I also wondered, and concluded the button was too far away when the text is short and the container is a little wide:

image

I think it looks better, but at the cost of the button being easily missed. With the button near the text it's not that nice, but it can't be missed:

image

There's also the alternative of making the banner not take the full width:

image

On the search page the container is smaller though, so it's different. Maybe we can have 2 different behaviors on the 2 pages.

Or we can always push it to the right. I don't have a strong opinion on that.

File: browser-extension/src/background.js [hunk 1]
```diff
@@ -342,6 +341,18 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
       process().then(sendResponse);
       return true;
     }
+  } else if (request.message === 'getBanners') {
+    const process = async () => {
+      const path = `backoffice/banners/`;
```
[the remark this answers is not in the document]
Comment (sigmike) [reply browser-extension/src/background.js]: Indeed. Maybe we should make it an error or warning on the linter. I just tried to configure the linter for that but strangely I had to remove the `prettier/prettier` rule to make it work, not sure why. This configuration should probably be revisited.

Done in 3e66646e9758fba0fc2cc9b99b8d8c2ad055134a (along with another one I found with the linter).

File: browser-extension/src/images/warning.svg
[the remark this answers is not in the document]
Comment (sigmike) [reply browser-extension/src/images/warning.svg:1]: I couldn't find the source of the other icons so I took the one used on the front. Done in 9dd134e31b8a052e0bdb40b4de4ea0c4e25f3436.

File: browser-extension/src/models/banner/Banner.js [hunk 2]
```diff
@@ -12,12 +9,26 @@ import { defaultBannerOptions } from './BannerOptions.js';
  * - etc.
```
[the remark this answers is not in the document]
Comment (sigmike) [reply browser-extension/src/models/banner/Banner.js]: Done in 9e7561b8291347ab0e0da56484248b57c822fa5e.

**Review by GresilleSiffle — APPROVED** [review]

**Pull request merged**